internal/querynode/impl.go

```diff
@@ -697,16 +712,28 @@ func (node *QueryNode) Search(ctx context.Context, req *queryPb.SearchRequest) (
 		failRet.Status.Reason = err2.Error()
 		return failRet, nil
 	}
+
+	failRet.Status.ErrorCode = commonpb.ErrorCode_Success
+	latency := tr.ElapseSpan()
+	metrics.QueryNodeSQReqLatency.WithLabelValues(fmt.Sprint(Params.QueryNodeCfg.GetNodeID()), metrics.SearchLabel).Observe(float64(latency.Milliseconds()))
+	metrics.QueryNodeSQCount.WithLabelValues(fmt.Sprint(Params.QueryNodeCfg.GetNodeID()), metrics.SearchLabel, metrics.SuccessLabel).Inc()
 	return ret, nil
 }
 
 // Query performs replica query tasks.
 func (node *QueryNode) Query(ctx context.Context, req *queryPb.QueryRequest) (*internalpb.RetrieveResults, error) {
+	metrics.QueryNodeSQCount.WithLabelValues(fmt.Sprint(Params.QueryNodeCfg.GetNodeID()), metrics.QueryLabel, metrics.TotalLabel).Inc()
 	failRet := &internalpb.RetrieveResults{
 		Status: &commonpb.Status{
 			ErrorCode: commonpb.ErrorCode_UnexpectedError,
 		},
 	}
+
+	defer func() {
+		if failRet.Status.ErrorCode != commonpb.ErrorCode_Success {
```

Comment:
`failRet.Status.ErrorCode` is always `UnexpectedError` in FromShardLeader branch